examples/CivisML_parallel_training.ipynb

```diff
@@ -0,0 +1,682 @@
+    "            'n_estimators': [50, 200],\n",
+    "            'max_depth': [1, 2]}\n",
+    "model = ModelPipeline(model='gradient_boosting_classifier',\n",
+    "                      dependent_variable=\"is_cancer\",\n",
+    "                      cross_validation_parameters=cv_params,\n",
+    "                      model_name='best_gbc_from_hyperparam_search')\n",
+    "models.append(model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Creating your own estimator"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Last but not least, I might want to roll my own classifier, apart from CivisML builtins. The CivisML docker imager allows me to insert my own classifiers from scikit-learn and other popular open source packages. You can see a list of them [here](https://civis-python.readthedocs.io/en/v1.5.2/ml.html#custom-models). It's easy to take your existing code and still get the benefits of CivisML parallelizability and metrics reporting.\n",
+    "\n",
+    "I've been playing around a lot recently with xgboost, let's create and parameterize our own XGBClassifier."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "xgb = xgboost.XGBClassifier(n_estimators=100, learning_rate=0.01)\n",
+    "model = ModelPipeline(model=xgb,\n",
+    "                      dependent_variable='is_cancer',\n",
+    "                      model_name='xgb_classifier')\n",
+    "models.append(model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "Great! Let's train several models. CivisML makes it really simple to kick off many parallel jobs. Creating a ModelPipeline is much like creating a scikit-learn model, in that it doesn't cause anything to happen computationally until the user calls \"train\" or \"predict.\"\n",
+    "\n",
+    "The futures.wait function will block until all models finish training. This might take a few minutes. Time for a coffee, maybe?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 756 ms, sys: 70.1 ms, total: 826 ms\n",
+      "Wall time: 38.3 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time \n",
+    "future_jobs = [model.train(df=df) for model in models]\n",
+    "futures.wait(future_jobs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The output of model.train() is a futures object. Calling .succeeded() on it will allow me to check the status of my job.\n",
+    "\n",
+    "Let's make sure that the state of each training job is \"succeeded.\" Jobs might fail, for example, if we drop a connection with the server. If this is the case, we can simply rerun the pipeline."
```

Comment:
If we lose our local connection, the job will keep running in the platform. In that case, you probably would sit in the `wait` call forever, since we'd never get the job completion notification.